### PR TITLE
DEV/DOC: remove latex/pdf documentation

### LINKED ISSUE #15635: CI:ASK: Remove LaTeX doc builds?
In every CI run we are firing up a LaTeX run for a document with more than 3K pages with questionable quality. I think nobody needs that file and for offline needs HTML docs are still a better option. But we waste a significant effort on CI machines for this file both in terms of energy and time. Design wise it's not a special document either and it's kind of an eyesore

Horrifying option of printing the docs entirely is, I'm hoping, not a thing at this point. And for specific parts one can still print out the html files. 

Any arguments against removal from CI jobs & artifacts?

### PR DESCRIPTION
**Pull request by tupui**
Closes gh-15635 and supersede part of gh-13917

* Remove the PDF documentation compilation from the CI.
* Remove all reference in doc and code to latex and PDF documentation (this is a separate single commit, @tylerjereddy please have a look since it's touching the paver file, I don't want to break things on your side)
* Refactor the CircleCI configuration to use commands.
* Add `[skip circle]` and `[circle skip]`
* Fix 2 doctests
* Version name in the doc is the name of the PR (does not affect builds for `main`)

### CODE REVIEW

File: scipy/signal/windows/_windows.py
Comment (larsoner) [inline scipy/signal/windows/_windows.py:339]: I would use the `with np.errstate(...)` context manager for these, it's nicer to contain the effect and make clear where/why it's needed

**Review by larsoner — APPROVED** [review]
Looks like a nice, clean simplification/removal! +1 for merge with or without my suggestion.

File: scipy/signal/windows/_windows.py
Comment (ilayn) [reply scipy/signal/windows/_windows.py:339]: There is/was #15333 about this  but I think we should make a better example indeed.

Comment (ilayn) [reply scipy/signal/windows/_windows.py:339]: There is/was #15333 about this  but I think we should make a better example indeed.

Comment (tupui) [reply scipy/signal/windows/_windows.py:339]: @ilayn indeed did not see this issue! Do you have something in mind? Or shall I just change the `freq` linspace to `np.linspace(-0.45, 0.45, len(A))` or something like this?

**Review by tylerjereddy — COMMENTED** [review]
Sounds good, just a few suggestions:

- in `doc/Makefile` do you mind getting rid of the upload step that moves the PDF on the doc server since the PDF won't exist anymore:
image
- if folks want to merge this and then open an issue for me to practice a few release commands later on I'd be okay with that, just ping me there or I can open the issue myself.. I probably can't practice the server upload but can cycle through `paver` and `make dist` maybe

**Comment by tupui** [comment]
> * in `doc/Makefile` do you mind getting rid of the upload step that moves the PDF on the doc server since the PDF won't exist anymore:

Good catch, thanks!

> * if folks want to merge this and then open an issue for me to practice a few release commands later on I'd be okay with that, just ping me there or I can open the issue myself.. I probably can't practice the server upload but can cycle through `paver` and `make dist` maybe

I will make this issue for you once this merge. Thanks @tylerjereddy

**Comment by tupui** [comment]
I "fixed" the skip logic to make it return a green CI.

@ilayn @larsoner for the doctests, what do you think about doing this instead `window = signal.windows.cosine(51)[1:-1]`? This would remove the need of silencing the error warning. And if ones want to improve further these examples it can be done later.

**Comment by tylerjereddy** [comment]
Ok, there is a core developer approval already, and I don't see anything that looks too suspicious in the diff. I guess the CI is being skipped at the moment so I can't quite tell if there are any final issues in CircleCI? I guess we'll find out soon!

I'll open the issue for me to check some things now. I'll keep the separate commits because if an issue is discovered during the release process that may help with dissection.

**Comment by tupui** [comment]
Thank you everyone, I will continue with the Meson PR then!

**Pull request merged**
